Fix unexpected argument autodetection_retry

Joe-Klauza opened the pull request:
Solves the following failure on startup:
```
python.d WARNING: plugin[main] : sqlserver[job_name] : unhandled exception on check : TypeError("connect(
) got an unexpected keyword argument 'autodetection_retry'",), skipping the job
```

Since we are updating `params` with `self.configuration` before calling `pymssql.connect(**params)`, we are adding `autodetection_retry` as an extraneous keyword argument. This causes an unhandled exception, preventing startup.

Commit:
Fix unexpected argument autodetection_retry

Solves the following failure on startup:
```
python.d WARNING: plugin[main] : sqlserver[job_name] : unhandled exception on check : TypeError("connect(
) got an unexpected keyword argument 'autodetection_retry'",), skipping the job
```

Since we are updating `params` with `self.configuration` before calling `pymssql.connect(**params)`, we are adding `autodetection_retry` as an extraneous keyword argument. This causes an unhandled exception, preventing startup.

liepumartins commented:
Thanks for noticing. After a quick glance, I believe it would be more error prone to create `params` using only required keys via something like 
```
params = {
    'host': conf.get('host'),            
    'database': conf.get('database'),    
}
```

If you could just adjust Your PR to this approach, I would be happy to quickly merge it.

EDIT: Just to clarify, I am currently away from my development system, and would not trust myself simply editing this via web, without testing.

Commit:
Connect via pymssql with explicit params

Only use explicit parameters (rather than everything left in the configuration) when connecting via pymssql.
